src/CheckableCell.js

```diff
@@ -1,98 +1,43 @@
-  View,
-  ViewPropTypes,
-  TouchableOpacity,
-  TouchableWithoutFeedback,
-} from 'react-native';
-
-/**
- * Renders a CheckableCell that renders either renderIsChecked or renderIsNotChecked when isChecked
- * is true or false respectively. Whole cell returned is pressable. Callback should affect state of
- * Parent in some way that keeps the state of parent in sync with state of the CheckableCell. Kept
- * separate to maintain responsiveness of the cell.
- * @param   {object}  props             Properties passed where component was created.
- * @prop    {StyleSheet} style          Style of the CheckableCell (View props).
- * @prop    {number} width              Flexbox flex property, gives weight to the CheckableCell width
- * @prop    {object} renderIsChecked    Object is rendered as child in CheckableCell if checked.
- * @prop    {object} renderIsNotChecked Object is rendered as child in CheckableCell if notchecked.
- * @prop    {boolean} isChecked         Used to set the initial state of the cell when the
- *                                      component mounts or rerenders (e.g. table sort
- *                                      order change).
- * @return  {React.Component}           Return TouchableOpacity with child rendered according to the
- *                                      above 3 props.
- */
-export class CheckableCell extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      isChecked: props.isChecked,
-    };
-    this.onPress = this.onPress.bind(this);
-  }
-
-  componentWillReceiveProps(nextProps) {
-    if (nextProps.isChecked !== this.state.isChecked) {
-      this.setState({ isChecked: nextProps.isChecked });
-    }
-  }
-
-  onPress() {
-    this.setState({ isChecked: !this.state.isChecked });
-    this.props.onPress();
-  }
-
-  render() {
-    const {
-      isDisabled,
-      style,
-      width,
-      renderDisabled,
-      renderIsChecked,
-      renderIsNotChecked,
-    } = this.props;
-
-    if (isDisabled) {
-      let renderFunction = renderDisabled;
-      if (!renderFunction) {
-        renderFunction = this.state.isChecked ? renderIsChecked : renderIsNotChecked;
-      }
-      return (
-        // Filler function for onPress stops press through to parent component
-        <TouchableWithoutFeedback onPress={() => {}}>
-          <View style={[style, { flex: width }]}>{renderFunction()}</View>
-        </TouchableWithoutFeedback>
-      );
-    }
+import React from 'react'
+import { TouchableOpacityPropTypes } from 'react-native'
+
+import { TouchableCell } from './TouchableCell'
+
+export const CheckableCell = React.memo(
+  ({
+    rowKey,
+    columnKey,
+    isChecked,
+    CheckedComponent,
+    UncheckedComponent,
+    disabled,
+    onCheckAction,
+    onUncheckAction,
+    dispatch,
+  }) => {
+    console.log(`- CheckableCell: ${rowKey},${columnKey}`)
+
+    const _onPressAction = isChecked ? onUncheckAction : onCheckAction
+
+    const _renderCheck = () =>
```

Comment:
I think you'd use the same component, just a prop `checked` where styles change/component renders differently? And if you do that is this needed or can you just use `TouchableCell`?

Reply:
I was actually planning to go ham and make it have 4 component props:
- Checked...
- Unchecked...
- CheckedDisabled...
- UncheckedDisabled...

This is just a helper cell for a common use case in mobile. It should handle the 4 render states you could expect. You could equally make the argument that it's redundant along with the rest of this package, cause you can just implement with VirtualisedList, View, TextInput, TouchableOpacity, Text and flexibility of anything else. The base components are just some default styled sugar.